frontend/src/pages/auth/PasswordChange.tsx

```diff
@@ -21,67 +21,24 @@ export default function PasswordChangePage() {
   // Toast is used for showing success/error messages
   const toast = useRef<Toast>(null);
 
-  const showSuccess = () => {
-    toast.current?.show({ severity: "success", summary: "Password Changed" });
-  };
-
-  const showFailure = (message: string) => {
-    toast.current?.show({ severity: "error", summary: message });
-  };
-
-  const header = <div className="font-bold mb-3">Pick a password</div>;
-  const footer = (
-    <>
-      <Divider />
-      <p className="mt-2">Suggestions</p>
-      <ul className="pl-2 ml-2 mt-0 line-height-3">
-        <li>At least one lowercase</li>
-        <li>At least one uppercase</li>
-        <li>At least one numeric</li>
-        <li>Minimum 8 characters</li>
-      </ul>
-    </>
-  );
-
   const onSubmit = (event: FormEvent<HTMLFormElement>): void => {
     logger.debug("Password Change Clicked");
-    if (newPassword1 === newPassword2) {
-      if (newPassword1.length > 7 && newPassword2.length > 7) {
-        if (
-          containsUppercase(newPassword1) &&
-          containsUppercase(newPassword2)
-        ) {
-          if (
-            containsLowercase(newPassword1) &&
-            containsLowercase(newPassword2)
-          ) {
-            if (containsNumber(newPassword1) && containsNumber(newPassword2)) {
-              AUTH_API.passwordChange({
-                old_password: oldPassword,
-                password: newPassword1,
-                password2: newPassword2,
-              }).then((response) => {
-                // TODO: Refactor this
-                if (response.data?.status) {
-                  showSuccess();
-                } else {
-                  showFailure("Error");
-                }
-              });
-            } else {
-              showFailure("New Password does not contain number");
-            }
-          } else {
-            showFailure("New Password does not contain lowercase letter");
-          }
+    const pwCheckRet = passwordValueCheck(newPassword1, newPassword2);
+    if (pwCheckRet[0]) {
+      AUTH_API.passwordChange({
+        old_password: oldPassword,
+        password: newPassword1,
+        password2: newPassword2,
+      }).then((response) => {
+        // TODO: Refactor this
+        if (response.data?.status) {
+          showSuccess(toast, "Password Changed");
         } else {
-          showFailure("New Password does not contain uppercase letter");
+          showFailure(toast, "Error");
```

Comment:
I think this is old code, and can be changed to .then().catch like everything else, rather than if/else. We'll have to check with Hosung, but I think from now on we should also be able to display error.data.errors[0] based on the API contract, but not 100% sure

Reply:
If this is not the case, we should ask to change it so that the error message is informative